src/Packagist/Api/Client.php

```diff
@@ -98,23 +99,25 @@ public function get(string $package)
     }
 
     /**
-     * Similar to {@link get()}, but uses composer metadata which is Packagist's preferred
-     * way of retrieving details, since responses are cached efficiently as static files
-     * by the Packagist service. The response lacks some metadata that is provided
-     * by {@link get()}, see https://packagist.org/apidoc for details.
-     *
-     * Caution: Returns an array of packages, you need to select the correct one
-     * from the indexed array.
-     *
-     * @since 1.6
-     *
+     * @see https://packagist.org/apidoc#get-package-data
+     * Contains tagged releases and dev versions
      * @param string $package Full qualified name ex : myname/mypackage
-     *
-     * @return \Packagist\Api\Result\Package[] An array of packages, including the requested one.
+     * @return Package[] An array of packages, including the requested one.
+     */
+    public function getComposer(string $package): array
+    {
+        return $this->respond(sprintf($this->url('/p2/%s~dev.json'), $package));
```

Comment:
Be careful. The `~dev.json` files is a not about getting all versions. It is about getting dev versions. None of the releases (be them stable releases or beta releases) are in that file. If you want to get all versions, you need to load *both* files.